### ISSUE TITLE
Unable to load a saved pytorch model due to trainer being required [BUG]

### ISSUE THREAD

**Issue by asross** [unit8co/darts#1116]
**Describe the bug**

When trying to run `NBEATSModel.load_model` on a file I saved using `save_model`, I get the following error:

```
RuntimeError: _NBEATSModule is not attached to a `Trainer`.
```

**To Reproduce**

```python
model = NBEATSModel(**my_params)
model.fit(my_data)
model.save('my_path.pth.tar')
NBEATSModel.load_model('my_path.pth.tar')
```

**Expected behavior**

The model should get successfully loaded and not raise any errors.

**System (please complete the following information):**
 - Python version: 3.9.13
 - darts version 0.20.0
 - MacOS 12.4

**Additional information**

Relevant part of the traceback:

```
  File "<PYTHON>/site-packages/darts/models/forecasting/torch_forecasting_model.py", line 1331, in load_model
    model.trainer = model.model.trainer
  File "<PYTHON>/site-packages/pytorch_lightning/core/module.py", line 179, in trainer
    raise RuntimeError(f"{self.__class__.__qualname__} is not attached to a `Trainer`.")
```

**Comment by aloysiogl** [comment]
Same issue here but with RNNModel, the issue seems to come from the latest version of pytorch-lightning. My temporary workaround was downgrading to `pytorch-lightning==1.6.5`.

**Comment by asross** [comment]
Cool, thanks for the suggestion :) My temporary workaround was just to use `load_from_checkpoint` instead, which doesn't trigger this error.

**Comment by dennisbader** [comment]
Thanks for that. Should be fixed with #1124 in the coming release
